### PR TITLE
List all VAST options in example vast.conf

### PR DESCRIPTION
**Pull request by dominiklohmann**

### CODE REVIEW

File: vast.conf [hunk 1]
```diff
@@ -1,116 +1,380 @@
...
-;; The host and port to listen at and connect to.
-; endpoint = "localhost:42000"
+  ; The host and port to listen at and connect to.
+  ;endpoint = "localhost:42000"
+
+  ; The file system path used for persistent state.
+  ;db-directory = "vast.db/"
+
+  ; The file system path used for log files.
+  ;log-file = "vast.db/server.log"
```
Comment (mavam) [inline vast.conf]: Is this taking the variable `db-directory` or the hard-coded string `vast.db`?

Comment (dominiklohmann) [reply vast.conf]: It's `<db-directory>/server.log` by default. I should write it like that.

**Comment by dominiklohmann** [comment]
@mavam do you consider this worth a changelog-entry? I don't know whether people are looking at the example configuration regularly.

**Comment by mavam** [comment]
> do you consider this worth a changelog-entry?

No, it's just a small change in logging output.

**Comment by dominiklohmann** [comment]
> No, it's just a small change in logging output.

You're thinking of the wrong PR, this is actually just a change in the example configuration.

**Comment by mavam** [comment]
Yes, sorry, my tabs got messed up. 😅 

I would say this one can also go without changelog entry.

File: vast.conf [hunk 2]
```diff
@@ -1,116 +1,380 @@
...
-;; The host and port to listen at and connect to.
-; endpoint = "localhost:42000"
+  ; The host and port to listen at and connect to.
+  ;endpoint = "localhost:42000"
+
+  ; The file system path used for persistent state.
+  ;db-directory = "vast.db/"
+
+  ; The file system path used for log files.
+  ;log-file = <server.log in db-directory>
```
Comment (mavam) [inline vast.conf]:
```suggestion
  ;log-file = "${db-directory}/server.log"
```

This seems a bit less ambiguous IMO. Maybe `<db-directory>/server.log`, but I would not combine text and variable in that fashion.

File: vast.conf [hunk 3]
```diff
@@ -1,116 +1,380 @@
...
+
+  ; Number of events to be packaged in a table slice (this is a target value that
```
Comment (mavam) [inline vast.conf]:
```suggestion
  ; Number of events to be batched in a table slice (this is a target value that
```

File: vast.conf [hunk 4]
```diff
@@ -1,116 +1,380 @@
...
+  ; can be underrun if the source has a low rate).
+  ;table-slice-size = 100
+
+  ; The table slice type (default|arrow).
```
Comment (mavam) [inline vast.conf]:
```suggestion
  ; The table slice type (default|arrow|msgpack).
```

It's okay to include his in here, even if the OSS version doesn't have it. When selecting this, the error message should "this is a pro feature".

File: vast.conf [hunk 5]
```diff
@@ -1,116 +1,380 @@
...
+  ;table-slice-type = "default"
+
+  ; The size of an index shard.
+  ;max-partition-size = 1000000
+
+  ; The unique ID of this node.
+  ;node-id = "node"
```
Comment (mavam) [inline vast.conf]:
```suggestion
  ;node-id = "vast"
```

Does this make sense?

File: vast.conf (hunk 2 above)
Comment (dominiklohmann) [reply vast.conf]: The angle brackets are consistent with the style used by CAF for values that are determined at runtime. The comment at the very top of the file explains what the angle brackets mean.

File: vast.conf (hunk 5 above)
Comment (dominiklohmann) [reply vast.conf]: The default value is node. This PR merely documents the default values.

File: vast.conf (hunk 2 above)
Comment (mavam) [reply vast.conf]: Okay, but is `in` a keyword? that confuses me. Having a `/` would make things more readable IMO.

Comment (dominiklohmann) [reply vast.conf]: As in `<db-directory/server.log>`?

Comment (dominiklohmann) [reply vast.conf]:
```suggestion
  ;log-file = "<db-directory>/server.log"
```

File: vast.conf (hunk 4 above)
Comment (dominiklohmann) [reply vast.conf]: Let's discuss this in tomorrow's standup, as there are some additional options in commands in VAST Pro that are not mentioned here at all yet.

Comment (dominiklohmann) [reply vast.conf:20]: I think it's better not to block this PR from being merged because of this. I'm resolving this without adding `msgpack`—for now.

**Pull request merged**